### ISSUE TITLE
Develop labeling workflow

### ISSUE THREAD

**Issue by katie-lamb** [catalyst-cooperative/mozilla-sec-eia#15]
Update 4/21:

I was able to use Label Studio to fully label one of the example docs. This is the existing workflow:
* Create a PDF of a filing using the utility function `save_as_pdf`. We need to setup a database that tracks which CIK's are part of the training dataset
* Use [this notebook](https://github.com/catalyst-cooperative/mozilla-sec-eia/blob/extraction-scoping/notebooks/07-kl-label-studio-json-creation.ipynb) to take a PDF filing, create bounding boxes around each word in the document, and create the necessary inputs for Label Studio: 
  * an image of the Ex. 21 page
  * JSON that's formatted as a Label Studio "task" with annotations for bounding boxes 
* Upload the image and JSON task file to the `labeled-ex21-filings` GCS bucket
* Sync Label Studio with GCS
* Manually label the entity for words in the Ex. 21 table (default value is no-object)
* Export the source JSON "task" so that the document could be re-imported into Label Studio, export a JSON in COCO format which also exports an image along with that JSON
* Run post-processing on the COCO format JSONs to create a training dataset in the format expected by LayoutLM

Original description:

We need to create a small training dataset that can be used to fine-tune an extraction model for the Ex. 21 tables. Ideally we'd use a UI like LabelStudio to do this, because it's easier to coordinate the effort, but we could create a notebook that walks through the labeling process. I labeled two examples in a notebook and it was kind of slow but doable. I also experimented with using the Label Studio OCR template, but drawing bounding boxes was slow and not normalized. It seems like it's basically necessary to use one of the Label Studio templates (because configuring a custom one is confusing), but maybe we could figure out how to import bounding boxes into the OCR template.

* The training dataset should be formatted like [this one](https://huggingface.co/datasets/nielsr/funsd-layoutlmv3).
* See [this notebook](https://github.com/catalyst-cooperative/mozilla-sec-eia/blob/extraction-scoping/notebooks/05-kl-training-data-creation.ipynb) for a disorganized example of creating training data and fine-tuning LayoutLM

**Labels:**
Read about [Inside-Outside-Beginning Format](https://en.wikipedia.org/wiki/Inside%E2%80%93outside%E2%80%93beginning_(tagging)#:~:text=The%20IOB%20format%20(short%20for,named%2Dentity%20recognition)) to understand what the prefix for these labels mean.

* `B-Subsidiary`
* `I-Subsidiary`
* `B-Location`
* `I-Location`
* `B-Own_Percentage`
* `I-Own_Percentage`

**Questions:**
* It's possible to filter the SEC 10K filings by sector. Does it make sense to filter for electricity and gas filers so that we can create a dataset more specific for those companies? Or should we try and fully generalize the dataset?
* Do we need a "header" label? Can we do the extraction without knowing the headers? Probably fewer classes is better.

```[tasklist]
### Tasks
- [x] Create bounding boxes around each word in a PDF (not each phrase/line)
- [x] Fix `accelerate` dependency and make sure that a dataset with 2 records works for fine-tuning LayoutLM
- [x] Experiment with importing bounding boxes into Label Studio
- [x] Choose a Label Studio template
- [x] Resolve permissions issues with GCS so LabelStudio can import from GCS
- [x] Create a DB with the CIK's of the training set and test set + a workflow to add new CIK's to the train set, can this be done with MLFlow?
- [x] Filter metadata for 50 or so filings from utilities to use as training data and create Label Studio task JSONs for them. Filtering for electricity/gas filings requires creating a DB with basic company info
- [x] Create script to handle creation of PDFs for training data filings, creation of images and JSON tasks for label studio, uploading to GCS
- [x] Label a couple docs and then update the README with the labeling workflow
- [x] Rename output JSONs in GCS to have the input filename match
```
